### DISCUSSION TITLE
Bagaimana cara merubah label tampilan error seperti ini

### DISCUSSION THREAD

**Discussion by Nurfalah123** [teamalgoritma/community#110] [Q&A] [answered]
Izin bertanya Bpk/Ibu

Saya ingin merubah label eror seperti gambar terlampir, secara data sudah sesuai karena pada usia rentang 1-5 tahun tidak ada yang sudah masuk ke universitas. Jadi saya ingin merupakan kata-kata erornya (tampilan merah) menjadi misal: "Data Tidak Ditemukan".

Berikut kodenya:
  output$plot5 <- renderPlotly({
    
    gover <- student %>% 
      select(Adaptivity.Level, Institution.Type, Age, Education.Level, Gender) %>%
      filter(Institution.Type == "Government") %>%
      filter(Education.Level == input$input_edu) %>%
      filter(Age == input$input_age) %>% 
      filter(Gender == input$input_gender) %>% 
      group_by(Adaptivity.Level) %>% 
      summarise(count=n()) %>% 
      ungroup() %>% 
      mutate(label = glue(
        "Level Adaptasi: {Adaptivity.Level}
    Total: {count}"
      )) %>% 
      arrange(-count)
    
    plot5 <- ggplot(data = gover, mapping = aes(x = reorder(Adaptivity.Level, count), 
                                                y = count,
                                                text = label)) +
      geom_segment(aes(x=reorder(Adaptivity.Level, count), xend=reorder(Adaptivity.Level, count), y=0,yend=count), size = 1, color="blue") +
      geom_point(size = 2, color="orange") +
      coord_flip() +
      labs(x = NULL, 
           y = NULL,
           fill = NULL) +
      theme_minimal()
    
    
    ggplotly(plot5, tooltip = "text")
    
    
  })

Dan ini capture erornya:

![error](https://user-images.githubusercontent.com/103617918/168481214-f387e309-e7f8-4d6f-bb6d-cc67bdd6fc36.PNG)

**Comment by NabiilahArdini** [comment, marked as answer]
Halo @Nurfalah123 ,

Anda bisa menggunakan fungsi validate dari shiny: https://shiny.rstudio.com/reference/shiny/0.14/validate.html Berikut dokumentasinya: https://shiny.rstudio.com/articles/validation.html Fungsi tersebut akan memastikan suatu kondisi terpenuhi, dan bila tidak terpenuhi akan memberikan pesan error yang nyaman dibaca. 

Masukan fungsi validate beserta kondisinya di dalam fungsi `render*({___})` sebelum kode apapun di dalamnya. biasanya kondisi yang dimasukan berkaitan dengan input di shiny. Berikut contohnya:

```
trend_data <- renderPlotly({
  
  validate(
    need(input$status != "", "Please fill all inputs provided.")
  )
  
  retail_clean %>% 
    filter(status %in% input$status) %>% 
    ggplot(aes(x = Category, y = Product_Sold) + 
      geom_col()
  
})
```

Untuk code Anda, kemungkinan besar code validate nya seperti ini:

```
output$plot5 <- renderPlotly({

validate(
    need(input$input_edu != "University" & input$input_age != "1-5", "Data Tidak Ditemukan.")
  )

gover <- student %>% 
  select(Adaptivity.Level, Institution.Type, Age, Education.Level, Gender) %>%
  filter(Institution.Type == "Government") %>% 
  ....(dst.)

})
```
